.github/workflows/template-tests.yml

```diff
@@ -0,0 +1,102 @@
+name: Templates
+
+on:
+  pull_request:
+    branches:
+      - main
+      - 'release/**'
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  # Duplicated jobs so their dependencies are not blocked on both the
+  # setup jobs
+
+  setup_for_tests_lin:
+    name: Setup for tests (Linux)
+    runs-on: ubuntu-latest
+    outputs:
+      tests_matrix: ${{ steps.generate_test_matrix.outputs.tests_matrix }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: ./.github/actions/enumerate-template-tests
+        id: generate_test_matrix
+
+  setup_for_tests_win:
+    name: Setup for template tests (Windows)
+    runs-on: windows-latest
+    outputs:
+      tests_matrix: ${{ steps.generate_test_matrix.outputs.tests_matrix }}
+    steps:
+      - uses: actions/checkout@v4
+      - id: generate_test_matrix
+        uses: ./.github/actions/enumerate-template-tests
+
+  test_linux:
+    name: Linux
+    uses: ./.github/workflows/run-tests.yml
+    needs: setup_for_tests_lin
+    strategy:
+      fail-fast: false
+      matrix: ${{ fromJson(needs.setup_for_tests_lin.outputs.tests_matrix) }}
+    with:
+      testShortName: ${{ matrix.shortname }}
+      os: "ubuntu-latest"
+      testProjectPath: tests/Aspire.Workload.Tests/Aspire.Workload.Tests.csproj
+      testSessionTimeoutMs: 1200000
+      testHangTimeout: 12m
```

Comment:
With a 20 minute session timeout, is having 12 minute timeout for each test too much?